src/actions/index.js

```diff
@@ -1,7 +1,39 @@
+import { $signIn } from "../requests"
+import * as actionTypes from "./types"
+import { startLoading, stopLoading } from "./loading"
+
 let nextMessageId = 0
 
 export const addMessage = (text, owner) => ({
-  type: 'ADD_MESSAGE',
+  type: "ADD_MESSAGE",
+  id: nextMessageId++,
+  text,
+  owner
+})
+
+export const login = form => dispatch => {
+  dispatch(startLoading())
```

Comment:
With a control before if the app is already loading something, you can prevent your app to send multiple requests if some user press a lot of time on enter key